### ISSUE TITLE
Unhandled Exception in object.cpp when using GDscript data Resource in a scene and resource_local_to_scene=true

### ISSUE THREAD

**Issue by ChainedLupine** [godotengine/godot#43709]
**Godot version:**
Godot beta 2 of 3.2.4 -- hash 04103db6bd5694b81ab0a1717fc5fdde6cb5dd4f

**OS/device including version:**
Windows 10

**Issue description:**
If creating a Resource script which holds data, and then setting Local To Scene to true, a crash occurs when it is loaded.

I suspect this is related to the reference count changes made around here: #43049

Unhandled exception in object.cpp line 414 at this point:

```
void Object::set(const StringName &p_name, const Variant &p_value, bool *r_valid) {

#ifdef TOOLS_ENABLED

	_edited = true;
#endif

	if (script_instance) {

		if (script_instance->set(p_name, p_value)) {   // <-- Crash here, script_instance pointer looks corrupt, just like I was seeing in #43049
			if (r_valid)
				*r_valid = true;
			return;
		}
	}
```

**Steps to reproduce:**

1. Create a data Resource, then load it via another script.  Local To Scene = yes.
2. Project crashes when executed.

**Minimal reproduction project:**

[godot_3.2.4_beta2_resource_local_to_scene_crash.zip](https://github.com/godotengine/godot/files/5571540/godot_3.2.4_beta2_resource_local_to_scene_crash.zip)

**Comment by ChainedLupine** [comment]
Mentioning @RandomShaper just in case he has an insight.

**Comment by akien-mga** [comment]
I can confirm the crash on 3.2.4.beta2, but it seems fixed in 06389f705b93db80cf0467c0a0b0972b7fa49680.

It's the same bug as #43652 which was fixed by #43657 (and cherry-picked for `3.2` via ebdcf1cc601ad847b11a9e15e4c07a017c265dd6).

If you want to confirm, you can use the windows-editor artifact from the latest `3.2` commit: https://github.com/godotengine/godot/runs/1426905927 (see Artifacts menu in the top right corner for the download). Note that CI artifacts are bigger than official builds as they include debug symbols, and use less optimizations.

**Comment by ChainedLupine** [comment]
Thanks @akien-mga!  Confirmed fixed for both the test project and my main game project.
